### ISSUE TITLE
Release signing keys

### ISSUE THREAD

**Issue by maltfield** [sublinks/sublinks-api#382]
Please add the official sublinks release signing keys to this repo.

I recommend adding a `KEYS` file to the root of your repo (located along-side files like `LICENSE` and `README.md`), per the [KEYS standard established by Apache](https://infra.apache.org/release-signing#keys-policy)

 * https://infra.apache.org/release-signing#keys-policy

If sublinks doesn't yet a cryptographic keypair for signing releases, then please generate them.

**Comment by maltfield** [comment]
For more best-practices, see also:

1. https://infra.apache.org/release-signing
2. https://docs.opendev.org/opendev/system-config/latest/signing.html
3. https://wiki.debian.org/Subkeys
4. https://riseup.net/en/security/message-security/openpgp/best-practices

**Comment by lazyguru** [comment]
This can be done closer to MVP release. Right now we don't actually have *any* releases (only nightly/bleeding-edge builds) and no one should be running Sublinks in any kind of "production"/"live" state. 

But yes, we'll have stuff available for verification of binaries.

**Comment by maltfield** [comment]
Yep, just wanted to make sure this infrastructure and process got setup properly *before* the first release.

**Comment by lazyguru** [comment]
Just read the Lemmy issue. The one thing I will add here is that our preferred way for people to install Sublinks will be Docker. So while I'm not saying we won't do this issue, I am saying that we _might_ not support downloading the binaries and running the service in a way that aligns with what you are asking for unless Docker fixes their stuff.

**Comment by maltfield** [comment]
While I do think it's reasonable to provide an insecure-yet-convenient method to install your software, I think it's very important that you provide at least one way for users to obtain your software securely.

**Comment by lazyguru** [comment]
I'll clarify. When I say we might not support it, I mean that while we may generate and sign the binaries according to this feature request, if users have issues running the platform from those binaries our answer will likely be to tell them to use docker instead. 

(As bad as this next part can come off, I really don't mean it that way. Text communication is hard due to the lack of context clues).

This is an open source project that currently receives little to no funding. All issues are open to anyone to pickup (though I recognize this specific one will require at least SOME support from the core devs). We simply are not going to be equipped to provide support for users attempting to run the platform on any number of possible hardware or OS configurations. If someone is willing to write up instructions for getting things running on a specific OS and architecture we will gladly accept those PRs. But I also want to be realistic in that issues opened for problems running on configurations other than docker will likely go unanswered or receive a comment asking the reporter to use docker instead (we may even go as far as closing the issue and asking the reporter to ask for help in a community forum instead). So while I agree with the idea behind this feature request, in practice it may only end up being useful to a very small number of instance owners. (Again, I'm in favor of keeping this open and implementing it in the future. I'm just not sure it will actually solve any problems. The more realistic approach to solving the security problems presented here would be to push for one of the container platforms out there to fix the issue on their side allowing for us as well as others like Lemmy to be able to provide secure downloads using a convenient install/run mechanism)

**Comment by Pdzly** [comment]
We will be doing our best, to secure it. 

But for docker images, we can just note the SHA256 hash of the docker image that is "authentic" released by us.

**Comment by maltfield** [comment]
just to be clear: hashes do *not* provide any security/authenticity unless they're signed with a key whoose private key is stored offline.
